File: 1_Arrays/1_MajorityElement.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        majorityElementVal = nums[0]
        count = 0
        for n in range(len(nums)):
            if count == 0:
                majorityElementVal = nums[n]
            count += (1 if majorityElementVal == nums[n] else -1)

        return majorityElementVal

File: 1_Arrays/1_MajorityElement.java
class Solution {
    public int majorityElement(int[] nums) {
      int count = 0;
      int majorityElementVal = nums[0];
      for ( int i = 0; i < nums.length; i++)
      {
        if (count == 0)
        {
          majorityElementVal = nums[i];
        }
        if (majorityElementVal == nums[i])
        {
          count++;
        }
        else
        {
          count--;
        }
      }
      return majorityElementVal;
        
    }
}